addons/common/persistence/jdbc/src/main/java/org/kie/kogito/persistence/jdbc/JDBCProcessInstances.java

```diff
@@ -80,10 +94,34 @@ private void init() {
             return;
         }
 
-        try (Connection connection = dataSource.getConnection();
-                PreparedStatement statement = connection.prepareStatement(getQueryFromFile("exists_tables"))) {
+        try (Connection connection = dataSource.getConnection()) {
+            final DatabaseMetaData metaData = connection.getMetaData();
+            final String dbProductName = metaData.getDatabaseProductName();
+            DatabaseType databaseType = DatabaseType.create(dbProductName);
+            if (databaseType == null) {
+                throw new Exception("Database (" + dbProductName + ") not suported");
+            }
+            switch (databaseType) {
+                case ORACLE:
+                    repository = new OracleRepository();
+                    break;
+                case POSTGRES:
+                    repository = new PostgresRepository();
+                    break;
+            }
```

Comment:
Applying the changes I suggested this whole code becomes way simpler
```suggestion
            DatabaseType databaseType = DatabaseType.create(dbProductName);
            repository = databaseType.getRepository();
```